.github/workflows/test-suite.yml

```diff
@@ -2,24 +2,47 @@ name: 'Test Suite'
 
 on:
   workflow_dispatch:
+    inputs:
+      terraform_matrix:
+        description: 'Terraform matrix version (JSON array)'
+        default: '["1.6.x", "1.5.x", "1.4.x", "1.3.x", "1.2.x"]'
```

Comment:
I checked that this notation works, e.g. 1.4.x will run version 1.4.7. 
TF version list is here: https://releases.hashicorp.com/terraform/